### PR TITLE
add special-char

### LINKED ISSUE #1: Implement containsSpecialChar utility with configurable allowed special characters
**Description:**
A new utility function `containsSpecialChar(...)` is required to detect whether a given string contains special characters from a defined allowed set.

**Task**
Implement a function `containsSpecialChar(input: string)` that checks if the input string contains any special characters from the following set: `! @ # $ % ^ & * ( ) - _ + = ? . , ; :`

**Requirements**
- The function should return `true` if at least one character from the allowed special character set is present in the input string.
- Otherwise, it should return `false`.
- The implementation should be clean and reusable.

**Testing**
Add unit tests (minimum 4 cases), covering:
- String with no special characters
- String with one special character
- String with multiple special characters
- Edge cases (empty string / only special characters)

**Notes**
_Focus only on implementation and testing of the utility. Create the `feature/special-char` branch for this work._

### PR DESCRIPTION
**Pull request by ilhamuh97**

### CODE REVIEW

File: src/main/java/PasswordValidator.java [hunk 1]
```diff
@@ -60,11 +60,24 @@ public static boolean isCommonPassword(@NotNull String password) {
         return false;
     }
 
+    public static boolean containsSpecialChar(@NotNull String password, String allowed) {
```
Comment (emmanuel-santhosh) [inline src/main/java/PasswordValidator.java:63]: Elegant mit Annotation !

**Review by emmanuel-santhosh — APPROVED** [review]
looks good to me
